src/screens/Home/ResourcesItemScreen.js

```diff
@@ -78,37 +90,91 @@ export default function ResourcesItemScreen(props) {
       setType(resourceData.type);
       setSubheader(resourceData.subheader);
 
-      /* Different behavior if resource is of type STEPS or FREEFORM */
-      if (type === 'STEPS') {
-        firestore().collection('resource_items').doc(resourceID).collection('steps')
-          .get()
-          .then((stepsList) => {
-            const stepsData = stepsList.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
-
-            const sortedStepsData = stepsData.sort((a, b) => {
-              if (a.stepNumber > b.stepNumber) {
-                return 1;
+      /* Query all map markers */
+      firestore().collection('resource_items').doc(resourceID).collection('map_markers')
+        .get()
+        .then((markersList) => {
+          const markersData = markersList.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+
+          if (markersData.length) {
+            /* Calculates the average latitude and longitude to give the center of all markers */
+            const totalCoord = markersData.reduce((total, amount) => ({
+              latitude: total.latitude + amount.latitude,
+              longitude: total.longitude + amount.longitude,
+            }));
+
+            const avgCoord = {
+              latitude: totalCoord.latitude / (markersData.length),
+              longitude: totalCoord.longitude / (markersData.length),
+            };
+
+            /* Calculates maximum distance between center and a marker to size map deltas */
+            markersData.unshift({ latitude: 0, longitude: 0 });
```

Comment:
So this is still a side effect because the array is being modified. I guess if there's no way getting around that, I would prefer to use `let` so it's clear that we're being intentional with the side effects.